src/riak_cs_storage_mr.erl

```diff
@@ -0,0 +1,276 @@
+                     %% Because user accessible active manifest had
+                     %% been removed `sum_objs/2', active's here are
+                     %% invisible for users.
+                     add_to(Sum, #sum.active_invisible, CBB);
+                 {mp, writing, CBB} ->
+                     %% MP writing is visible for user. Also add
+                     %% to MP writing counters.
+                     Sum1 = add_to(Sum, #sum.user, CBB),
+                     add_to(Sum1, #sum.writing_multipart, CBB);
+                 {non_mp, writing, CBB} ->
+                     case new_or_old(LeewayEdge, M?MANIFEST.write_start_time) of
+                         new ->
+                             add_to(Sum, #sum.writing_new, CBB);
+                         old ->
+                             add_to(Sum, #sum.writing_old, CBB)
+                     end;
+                 {_, pending_delete, CBB} ->
+                     case new_or_old(LeewayEdge, M?MANIFEST.delete_marked_time) of
+                         new ->
+                             add_to(Sum, #sum.pending_delete_new, CBB);
+                         old ->
+                             add_to(Sum, #sum.pending_delete_old, CBB)
+                     end;
+                 {_, scheduled_delete, CBB} ->
+                     case new_or_old(LeewayEdge, M?MANIFEST.delete_marked_time) of
+                         new ->
+                             add_to(Sum, #sum.scheduled_delete_new, CBB);
+                         old ->
+                             add_to(Sum, #sum.scheduled_delete_old, CBB)
+                     end
+             end,
+    sum_objs(LeewayEdge, NewSum, Rest).
+
+-spec bytes_and_blocks(lfs_manifest()) ->
+                              {non_mp | mp, State::atom(),
+                               {Count::non_neg_integer(),
+                                Bytes::non_neg_integer(),
+                                Blocks::non_neg_integer()}}.
+bytes_and_blocks(?MANIFEST{props=Props} = M) when is_list(Props) ->
+    case proplists:get_value(multipart, Props) of
+        ?MULTIPART_MANIFEST{} = MpM -> bytes_and_blocks_mp(M, MpM);
+        _ -> bytes_and_blocks_non_mp(M)
+    end;
+bytes_and_blocks(M) ->
+    bytes_and_blocks_non_mp(M).
+
+bytes_and_blocks_non_mp(?MANIFEST{state=State, content_length=CL, block_size=BS})
+  when is_integer(CL) andalso is_integer(BS) ->
+    BlockCount = riak_cs_lfs_utils:block_count(CL, BS),
+    {non_mp, State, {1, CL, BlockCount}};
+bytes_and_blocks_non_mp(?MANIFEST{state=State} = _M) ->
+    lager:debug("Strange manifest: ~p~n", [_M]),
+    %% The branch above is for content_length is properly set.  This
+    %% is true for non-MP v2 auth case but not always true for v4 of
+    %% streaming sha256 check of writing. To avoid error, ignore
+    %% this objects. Can this be guessed better from write_blocks_remaining?
+    {non_mp, State, {1, 0, 0}}.
+
+%% There are possibility of understimatation and overestimation for
```

Comment:
What is these comments for? I'd recommend adding `@doc` at head, if this isdescribing `bytes_and_blocks_mp`.